test/e2e_node/container_lifecycle_test.go

```diff
@@ -5905,7 +5905,233 @@ var _ = SIGDescribe(feature.SidecarContainers, "Containers Lifecycle", func() {
+							}),
+							Lifecycle: makePreStop(restartableInit2, 10),
+						},
+						{
+							Name:          restartableInit3,
+							Image:         defaultImage,
+							RestartPolicy: &containerRestartPolicyAlways,
+							Command: ExecCommand(restartableInit3, execCommand{
+								Delay: 600, TerminationSeconds: 1, ExitCode: 0,
+							}),
+							Lifecycle: makePreStop(restartableInit3, 1),
+						},
+					},
+					Containers: []v1.Container{
+						{
+							Name:  regular1,
+							Image: defaultImage,
+							Command: ExecCommand(regular1, execCommand{
+								Delay: 600, TerminationSeconds: 1, ExitCode: 0,
+							}),
+						},
+					},
+				},
+			}
+
+			preparePod(pod)
+			client := e2epod.NewPodClient(f)
+
+			pod = client.Create(ctx, pod)
+			defer client.RemoveFinalizer(ctx, pod.Name, testFinalizer)
+
+			framework.ExpectNoError(
+				e2epod.WaitForPodRunningInNamespace(ctx, f.ClientSet, pod),
+			)
+
+			framework.ExpectNoError(
+				client.Delete(ctx, pod.Name, metav1.DeleteOptions{GracePeriodSeconds: &grace}),
+			)
+
+			framework.ExpectNoError(
+				e2epod.WaitForPodSuccessInNamespace(ctx, f.ClientSet, pod.Name, f.Namespace.Name),
+			)
+
+			pod, err := client.Get(ctx, pod.Name, metav1.GetOptions{})
+			framework.ExpectNoError(err)
+
+			expectPodTerminationContainerStatuses(pod.Status.ContainerStatuses, map[string]podTerminationContainerStatus{
+				regular1: {exitCode: 0, reason: "Completed"},
+			})
+			expectPodTerminationContainerStatuses(pod.Status.InitContainerStatuses, map[string]podTerminationContainerStatus{
+				restartableInit1: {exitCode: 0, reason: "Completed"},
+				restartableInit2: {exitCode: 137, reason: "Error"},
+				restartableInit3: {exitCode: 0, reason: "Completed"},
+			})
+		})
+	})
+
+	ginkgo.When("A restartable init container with startup probe fails initially", func() {
+		ginkgo.It("should continue probing and allow regular container to start after the restartable init recovers", func(ctx context.Context) {
+			restartableInit := "buggy-restartable-init"
+			regularContainer := "regular-container"
+
+			restartPolicyAlways := v1.ContainerRestartPolicyAlways
+
+			pod := &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "restartable-init-startup-probe-fix",
+				},
+				Spec: v1.PodSpec{
+					RestartPolicy: v1.RestartPolicyNever,
+					Volumes: []v1.Volume{{
+						Name: "workdir",
+						VolumeSource: v1.VolumeSource{
+							EmptyDir: &v1.EmptyDirVolumeSource{},
+						},
+					}},
+					InitContainers: []v1.Container{{
+						Name:          restartableInit,
+						Image:         busyboxImage,
+						RestartPolicy: &restartPolicyAlways,
+						VolumeMounts: []v1.VolumeMount{{
+							Name:      "workdir",
+							MountPath: "/work",
+						}},
+						Command: []string{"sh", "-c", "if [ ! -f /work/first_run_done ]; then echo 'First run: creating marker and exiting with 1'; touch /work/first_run_done; exit 1; else echo 'Second run: marker found, running as sidecar'; sleep 120; fi"},
```

Comment:
```suggestion
					InitContainers: []v1.Container{{
						Name:          restartableInit,
						Image:         busyboxImage,
						RestartPolicy: &restartPolicyAlways,
						Command: []string{"sh", "-c", `
if [ ! -f /persistent/first_run_done ]; then
  echo 'First run: creating marker and exiting with 1'
  touch /persistent/first_run_done
  exit 1
else
  echo 'Second run: marker found, running as sidecar'
  sleep 120
fi`},
```

1. use `/persistent` directory
2. use multilined string for readability